pkg/repository/v1_repository.go

```diff
@@ -174,42 +179,79 @@ func (r *V1Repository) updateComponentManifest(local v1manifest.LocalManifests,
 		}
 	}
 
-	err = local.SaveComponentManifest(manifest, filename)
+	err = r.local.SaveComponentManifest(manifest, filename)
 	if err != nil {
 		return nil, err
 	}
 
 	return &component, nil
 }
 
+// downloadComponent downloads a component using the relevant manifest and checks its hash.
+// Precondition: the component's manifest is up to date and the version and platform are valid.
+// The caller is responsible for closing the returned reader.
+func (r *V1Repository) downloadComponent(id string, platform string, version string) (io.Reader, error) {
+	manifest, err := r.local.LoadComponentManifest(v1manifest.ComponentFilename(id))
+	if err != nil {
+		return nil, err
+	}
+
+	item := manifest.Platforms[platform][version]
```

Comment:
Do we need to check whether `platform/version` exists?

Reply:
its up to the caller to ensure. I think the caller should know from their flow, so it would be a waste to check here (see also the precondition in the comment)